pkg/controllers/clusterresourceplacement/placement_status.go

```diff
@@ -215,40 +297,54 @@ func (r *Reconciler) buildClusterResourceBindings(ctx context.Context, crp *flee
 }
 
 // setResourcePlacementStatusPerCluster sets the resource related fields for each cluster.
-// It returns an array which records the status for each resource condition.
-// The resource condition order (index) is defined as const:
-// const (
-//
-//	RolloutStartedCondition resourceCondition = iota
-//	OverriddenCondition
-//	WorkSynchronizedCondition
-//	AppliedCondition
-//	AvailableCondition
-//	TotalCondition
-//
-// )
-func (r *Reconciler) setResourcePlacementStatusPerCluster(crp *fleetv1beta1.ClusterResourcePlacement, latestResourceSnapshot *fleetv1beta1.ClusterResourceSnapshot,
-	binding *fleetv1beta1.ClusterResourceBinding, status *fleetv1beta1.ResourcePlacementStatus) ([]metav1.ConditionStatus, error) {
+// It returns a map which tracks the set status for each relevant condition type.
+func (r *Reconciler) setResourcePlacementStatusPerCluster(
```

Comment:
the returned error is not needed any more?

Reply:
Ah, this is directly copied from the original code; it also always returns a nil error. Let me drop it.